controllers/operandrequest/reconcile_operand.go

```diff
@@ -805,89 +803,60 @@ func (r *Reconciler) createK8sResource(ctx context.Context, k8sResTemplate unstr
 	return nil
 }
 
-func (r *Reconciler) updateK8sResource(ctx context.Context, existingK8sRes unstructured.Unstructured, k8sResConfig []byte, newLabels, newAnnotations map[string]string) error {
+func (r *Reconciler) updateK8sResource(ctx context.Context, existingK8sRes unstructured.Unstructured, k8sResConfig *runtime.RawExtension, newLabels, newAnnotations map[string]string) error {
 	kind := existingK8sRes.GetKind()
 	apiversion := existingK8sRes.GetAPIVersion()
 	name := existingK8sRes.GetName()
 	namespace := existingK8sRes.GetNamespace()
 
-	// Update the k8s res
-	err := wait.PollImmediate(constant.DefaultCRFetchPeriod, constant.DefaultCRFetchTimeout, func() (bool, error) {
+	existingK8sRes = unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": apiversion,
+			"kind":       kind,
+		},
+	}
 
-		existingK8sRes := unstructured.Unstructured{
-			Object: map[string]interface{}{
-				"apiVersion": apiversion,
-				"kind":       kind,
-			},
-		}
+	if err := r.Client.Get(ctx, types.NamespacedName{
+		Name:      name,
+		Namespace: namespace,
+	}, &existingK8sRes); err != nil {
+		return errors.Wrapf(err, "failed to get k8s resource -- Kind: %s, NamespacedName: %s/%s", kind, namespace, name)
+	}
 
+	if !checkLabel(existingK8sRes, map[string]string{constant.OpreqLabel: "true"}) {
+		return nil
+	}
+
+	// delete the existing k8s resource, some resources could not be updated after it has been created
+	if err := r.Client.Delete(ctx, &existingK8sRes); err != nil {
```

Comment:
Will this line cause the k8s resource to get deleted in every reconcile loop?

Reply:
Yes, when ODLM is reconciling, and the resources is forced to update and label is correct. It will be deleted and re-created

Reply:
But even if we use the `Update()`, the resource will be updated in each reconcile loop except the resource could not be updated.

One solution is to compare the existing resource with the new template, and decide whether we should update it.

Reply:
Could I ask which k8s resouce, that can't be updated?

Reply:
like `job`, some of fields in `template` is immutable, so we could not update it.